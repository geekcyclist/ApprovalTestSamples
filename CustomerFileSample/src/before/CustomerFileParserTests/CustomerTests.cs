using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.Windows;
using Xunit;

namespace CustomerFileParserTests
{
    [UseReporter(typeof(BeyondCompare4Reporter))]
    [UseApprovalSubdirectory("Approvals")]
    public class CustomerTests
    {
        [Fact]
        public void CanPopulateCustomer()
        {
            int expectedId = 1;
            string expectedFirst = "Homer";
            string expectedMiddle = null;
            string expectedLast = "Simpson";
            string expectedStreet = "742 Evergreen Terrace";
            string expectedCity = "Springfield";
            string expectedState = "IL";
            string expectedZip = "62704";
            string expectedZip4 = "7023";
            int expectedAge = 39;

            var homer = MockData.CreateHomer();
            Assert.Equal(expectedId, homer.Id);
            Assert.Equal(expectedFirst, homer.FirstName);
            Assert.Equal(expectedMiddle, homer.MiddleName);
            Assert.Equal(expectedLast, homer.LastName);
            Assert.Equal(expectedStreet, homer.Street);
            Assert.Equal(expectedCity, homer.City);
            Assert.Equal(expectedState, homer.State);
            Assert.Equal(expectedZip, homer.Zip);
            Assert.Equal(expectedZip4, homer.Zip4);
            Assert.Equal(expectedAge, homer.Age);
        }

        [Fact]
        public void ApproveCustomer()
        {
            Approvals.Verify(MockData.CreateHomer());
        }
    }
}
