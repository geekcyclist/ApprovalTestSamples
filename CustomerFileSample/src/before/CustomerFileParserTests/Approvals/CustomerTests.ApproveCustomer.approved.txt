Id: 1
FirstName: Homer
MiddleName: (null)
LastName: Simpson
Suffix: (null)
Street: 742 Evergreen Terrace
City: Springfield
State: IL
Zip: 62704
Zip4: 7023
Age: 39
